src/project-manager.ts

```diff
@@ -306,10 +403,20 @@ export class ProjectManager implements Disposable {
 		throw new Error('unreachable');
 	}
 
+	/**
+	 * Called when file was opened by client. Current implementation
+	 * does not differenciates open and change events
+	 * @param fileName path to a file
```

Comment:
The method looks like the LSP method, but the LSP method takes a URI. You sure this is a path? If it's a path, is it relative to the root path?

(same for all `did*` methods)